### PR TITLE
Feat/eslintv9

### PR DESCRIPTION
**Pull request by Charlie-Crowdstrike**
* migrate `eslint-config-n` to `v17`
* use `eslintv9` version of `eslint-config-crowdstrike-node`
* migrate package to `eslintv9` flat config format

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,17 +34,19 @@
   "homepage": "https://github.com/CrowdStrike/eslint-config-crowdstrike#readme",
   "devDependencies": {
     "@crowdstrike/commitlint": "^7.0.0",
-    "eslint": "^8.0.0",
-    "eslint-config-crowdstrike": "link:",
```
Comment (kellyselden) [inline package.json:38]: Why was this link dropped?

File: index.js [hunk 2]
```diff
@@ -1,77 +1,87 @@
...
-    'no-restricted-globals': ['error', 'event'],
-    'object-shorthand': 'error',
-    'one-var': ['error', 'never'],
-    'quotes': ['error', 'single', {
-      'allowTemplateLiterals': true,
-      'avoidEscape': true,
-    }],
-    'require-await': 'error',
-    'semi': 'error',
+const js = require('@eslint/js');
```
Comment (kellyselden) [inline index.js:3]: Since we are directly importing this, it should be a real dep and not a peerDep.

File: package.json (hunk 1 above)
Comment (Charlie-Crowdstrike) [reply package.json:38]: Thanks for answering this in the sync, but for posterity:

new release of `eslintv9` compatible https://github.com/CrowdStrike/eslint-config-crowdstrike-node expects a `eslintv8` version of this repo, so pinning to the `eslintv8` version of this repo `@10.1.0`.

File: index.js (hunk 2 above)
Comment (Charlie-Crowdstrike) [reply index.js:3]: Good point, made an update to the other repo, https://github.com/CrowdStrike/eslint-config-crowdstrike-node/pull/95

**Review by kellyselden — APPROVED** [review]

**Comment by kellyselden** [comment]
+ eslint-config-crowdstrike@11.0.0-beta.0

**Pull request merged**